pypeit/core/pydl.py

```diff
@@ -1999,3 +2097,720 @@ def djs_laxisgen(dims, iaxis=0):
+                                while mapGroups[checkEarly] != checkEarly:
+                                    checkEarly = mapGroups[checkEarly]
+                                minEarly = min(minEarly, checkEarly)
+                            else:
+                                inGroup[self.chunkList[i][j][l]] = nMapGroups
+                            l = chunkGroup.nextGroup[l]
+                        if minEarly == 9*nPoints:
+                            mapGroups[nMapGroups] = nMapGroups
+                        else:
+                            mapGroups[nMapGroups] = minEarly
+                            l = chunkGroup.firstGroup[k]
+                            while l != -1:
+                                checkEarly = inGroup[self.chunkList[i][j][l]]
+                                while mapGroups[checkEarly] != checkEarly:
+                                    tmpEarly = mapGroups[checkEarly]
+                                    mapGroups[checkEarly] = minEarly
+                                    checkEarly = tmpEarly
+                                mapGroups[checkEarly] = minEarly
+                                l = chunkGroup.nextGroup[l]
+                        nMapGroups += 1
+        #
+        # Now all groups which are mapped to themselves are the real groups
+        # Make sure the mappings are set up to go all the way down.
+        #
+        nGroups = 0
+        for i in range(nMapGroups):
+            if mapGroups[i] != -1:
+                if mapGroups[i] == i:
+                    mapGroups[i] = nGroups
+                    nGroups += 1
+                else:
+                    mapGroups[i] = mapGroups[mapGroups[i]]
+            else:
+                msgs.error("MapGroups[{0:d}]={1:d} in chunks.friendsoffriends().".format(i, mapGroups[i]))
+        for i in range(nPoints):
+            inGroup[i] = mapGroups[inGroup[i]]
+        firstGroup = np.zeros(nPoints, dtype='i4') - 1
+        nextGroup = np.zeros(nPoints, dtype='i4') - 1
+        multGroup = np.zeros(nPoints, dtype='i4')
+        for i in range(nPoints-1, -1, -1):
+            nextGroup[i] = firstGroup[inGroup[i]]
+            firstGroup[inGroup[i]] = i
+        for i in range(nGroups):
+            j = firstGroup[i]
+            while j != -1:
+                multGroup[i] += 1
+                j = nextGroup[j]
+        return (inGroup, multGroup, firstGroup, nextGroup, nGroups)
+
+    def chunkfriendsoffriends(self, ra, dec, chunkList, linkSep):
+        """Does friends-of-friends on the ra, dec that are defined by
+        chunkList.
+        """
+        #
+        # Convert ra, dec into something that can be digested by the
+        # groups object.
+        #
+        x = np.deg2rad(np.vstack((ra[chunkList], dec[chunkList])))
+        radLinkSep = np.deg2rad(linkSep)
+        group = groups(x, radLinkSep, 'sphereradec')
+        return group
+
+
+class groups(object):
```

Comment:
Yes it is. I just had the code running in IDL with the IDL version and this is exactly the same. I suspect the pydl version has no advantages over scikit, but it would take some time to figure out. I've added this as a TODO.